src/main/java/by/itechart/api/controller/UserController.java

```diff
@@ -20,33 +22,32 @@
     private final UserServiceImpl userService;
 
     @GetMapping("/all")
+    @RolesAllowed("ROLE_ADMIN")
     public ResponseEntity<List<UserDTO>> getAllUsers() {
         return new ResponseEntity<>(userService.findAll(), HttpStatus.OK);
     }
 
-    /**
-     * status FOUND here is just for test. It will be changed later
-     * this method will get a authentication parameter later, but for now there will be no params
-     *
-     * @return current user
-     */
+    @RolesAllowed({"ROLE_ADMIN", "ROLE_USER"})
     @GetMapping("/current")
-    public ResponseEntity<UserDTO> getCurrentUser() {
-        return new ResponseEntity<>(HttpStatus.FOUND);
+    public ResponseEntity<UserDTO> getCurrentUser(Authentication authentication) {
+        return new ResponseEntity<>(userService.getCurrentUser(authentication), HttpStatus.FOUND);
     }
 
+    @RolesAllowed({"ROLE_ADMIN", "ROLE_USER"})
     @PatchMapping("{id}")
     public ResponseEntity<UserDTO> updateUser(@PathVariable Long id, @Valid @RequestBody UpdateUserDTO userDTO) {
         return new ResponseEntity<>(userService.update(id, userDTO), HttpStatus.OK);
     }
 
+    @RolesAllowed("ROLE_ADMIN")
     @DeleteMapping("{id}")
     public ResponseEntity<UserDTO> deleteUser(@PathVariable Long id) {
         userService.delete(id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @PostMapping
+    @RolesAllowed({"ROLE_ADMIN", "ROLE_USER"})
```

Comment:
As it was mentioned in the task description only `ADMIN` should be able to create a new user